collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java

```diff
@@ -746,6 +747,29 @@ protected void onBackPressedWithoutLogger() {
         super.onBackPressed();
     }
 
+    private void navigateToTheLastRelevantIndex(FormController formController) {
+        FormEntryController fec = new FormEntryController(new FormEntryModel(formController.getFormDef()));
+        formController.jumpToIndex(startIndex);
+
+        // startIndex might no longer exist if it was a part of repeat group that has been removed
+        while (true) {
```

Comment:
I don't love while(true), but the other option here would be a do-while and I like that even less so I agree with your choice.